src/util/sss_iobuf.c

```diff
@@ -19,18 +19,18 @@ struct sss_iobuf {
 };
 
 struct sss_iobuf *sss_iobuf_init_empty(TALLOC_CTX *mem_ctx,
-                                       size_t size,
-                                       size_t capacity)
+                                     size_t size,
+                                     size_t capacity)
 {
     struct sss_iobuf *iobuf;
     uint8_t *buf;
 
-    iobuf = talloc_zero(mem_ctx, struct sss_iobuf);
+    iobuf = talloc(mem_ctx, struct sss_iobuf);
     if (iobuf == NULL) {
         return NULL;
     }
 
-    buf = talloc_zero_array(iobuf, uint8_t, size);
+    buf = talloc_array(iobuf, uint8_t, size);
```

Comment:
I suggest to add bool arg to `sss_iobuf_init_empty()`  to determine if memory should be zeroed.

Reply:
All current calls would have this parameter set to `false` today. I think it will be better to add this parameter the day it is needed, if ever.

Reply:
Presence of this parameter warrants that user of this API actually thinks if he needs a clean buffer or garabage is ok.
I think it's better to have this in API. If not then at least a comment in the description of `size         The size of the data buffer` argument is required.

Reply:
Added the comments.

Reply:
I meant documentation in the header file:
https://github.com/SSSD/sssd/blob/a3ea7587794d12aa7f59b6bc7cab02ce4e648c07/src/util/sss_iobuf.h#L21

Reply:
Done